smithy-codegen-core/src/main/java/software/amazon/smithy/codegen/core/writer/CodegenWriter.java

```diff
@@ -80,10 +80,12 @@
  *
  * @param <T> The concrete type, used to provide a fluent interface.
  * @param <U> The import container used by the writer to manage imports.
+ * @deprecated prefer {@link SimpleCodegenWriter}
  */
 @SmithyUnstableApi
+@Deprecated
```

Comment:
Let's call out that we'll remove this in a future release since it's unstable.

Reply:
Done.